schemas/pa-yaml/v2.2/pa.schema.yaml

```diff
@@ -0,0 +1,350 @@
+
+$schema: http://json-schema.org/draft-07/schema#
+$id: http://powerapps.com/schemas/pa-yaml/v2.2/pa.schema
+title: Microsoft Power Apps schema for app source yaml files (v2.2).
+description: >-
+  The schema for all *.pa.yaml files which are used to describe a Power Apps canvas app.
+  All *.pa.yaml files in an *.msapp are logically combined into a single *.pa.yaml file.
+
+type: object
+additionalProperties: false
+properties:
+  App:
+    $ref: "#/definitions/App-instance"
+  Screens:
+    $ref: "#/definitions/Screens-name-instance-map"
+  ComponentDefinitions:
+    $ref: "#/definitions/ComponentDefinitions-name-instance-map"
+
+definitions:
+  App-instance:
+    description: Represents the top-level App entity.
+    type: object
+    additionalProperties: false
+    properties:
+      Properties: { $ref: "#/definitions/Properties-formula-map" }
+
+      # Note: App children is fixed.
+      Children:
+        description: The App currently only supports the 'Host' child.
+        type: object
+        additionalProperties: false
+        properties:
+          Host:
+            type: object
+            additionalProperties: false
+            properties:
+              # Currently, the Control identifier is static, but may need to be exposed in order to support variants
+              #Control: { $ref: "#/definitions/Control-type-identifier" }
+              Properties: { $ref: "#/definitions/Properties-formula-map" }
+
+  Screens-name-instance-map:
+    type: object
+    propertyNames: { $ref: "#/definitions/Screen-name" }
+    additionalProperties:
+      type: object
+      additionalProperties: false
+      properties:
+        Properties: { $ref: "#/definitions/Properties-formula-map" }
+        Children: { $ref: "#/definitions/Children-Control-instance-sequence" }
+
+  Screen-name:
+    description: The base requirements for a named entity in an app.
+    type: string
+    pattern: "^[a-zA-Z0-9_- ]+$"
+
+  Children-Control-instance-sequence:
+    description: >-
+      A sequence of control instances, where each item is a control's name with a control instance.
+      The order of the control instances in the sequence indicates the effective z-index of the control in the app.
+    type: array
+    items:
+      type: object
+      minProperties: 1
+      maxProperties: 1
+      propertyNames: { $ref: "#/definitions/Control-name" }
+      additionalProperties:
+        $ref: "#/definitions/Control-instance"
+
+  Control-name:
+    description: The base requirements for a named entity in an app.
+    type: string
+    pattern: "^[a-zA-Z0-9_].*$"
+
+  Control-instance:
+    allOf:
+      - $ref: "#/definitions/Control-instance-base"
+
+  Control-instance-base:
+    type: object
+    required: [Control]
+    properties:
+      Control: { $ref: "#/definitions/Control-type-identifier" }
+      Properties: { $ref: "#/definitions/Properties-formula-map" }
+      Children: { $ref: "#/definitions/Children-Control-instance-sequence" }
```

Comment:
Control has more items such as Invariant

Reply:
do you mean the `Variant`? yeah, I added this property.